### PR TITLE
SOAP test were added

### PR DESCRIPTION
**Pull request by kschernopiatova**

### CODE REVIEW

File: src/main/java/com/qaprosoft/carina/demo/soap/AddInteger.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import com.qaprosoft.carina.core.foundation.api.http.HttpResponseStatusType;
+import com.qaprosoft.carina.core.foundation.utils.Configuration;
+
+@Endpoint(url = "${base_url}", methodType = HttpMethodType.POST)
+@SuccessfulHttpStatus(status = HttpResponseStatusType.OK_200)
+@RequestTemplatePath(path = "api/soap/addinteger/rq.xml")
+@ResponseTemplatePath(path = "api/soap/addinteger/rs.xml")
+public class AddInteger extends AbstractApiMethodV2 {
+
+    public AddInteger() {
```
Comment (SergeiZagriychuk) [inline src/main/java/com/qaprosoft/carina/demo/soap/AddInteger.java:18]: also method name should include word "Method" like AddIntegerMethod

File: src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class SoapSampleTest implements IAbstractTest {
+
+    @Test
+    public void testLookupCity() {
+        LookupCity soap = new LookupCity();
+        soap.setProperties("api/soap/soap.properties");
+        soap.setHeaders(String.format("Content-type=%s", "text/xml"));
```
Comment (SergeiZagriychuk) [inline src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java:18]: I think by calling 'setHeaders' 2 times you will override your first setting. please review how setHeaders() method works.
Also there is annotation ContentType available for AbstractAPIMethod.
Please use it better instead

File: src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        soap.setHeaders(String.format("SOAPAction=%s", "http://tempuri.org/SOAP.Demo.LookupCity"));
+
+        Response response = soap.callAPI();
```
Comment (SergeiZagriychuk) [inline src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java:21]: to automatically check api status better use callAPIExpectSuccess()

File: src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        soap.setHeaders(String.format("Content-type=%s", "text/xml"));
+        soap.setHeaders(String.format("SOAPAction=%s", "http://tempuri.org/SOAP.Demo.AddInteger"));
+
+        Response response = soap.callAPI();
+        XmlPath rsBody = XmlPath.given(response.asString());
+        Integer actualResult = rsBody.getInt("AddIntegerResult");
+        Integer expectedResult = Integer.valueOf(soap.getProperties().getProperty("result"));
+
+        soap.expectResponseStatus(HttpResponseStatusType.OK_200);
+        Assert.assertEquals(actualResult, expectedResult);
```
Comment (SergeiZagriychuk) [inline src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java:43]: in one of tests use validateResponse() method instead

File: src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java (hunk 2 above)
Comment (SergeiZagriychuk) [reply src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java:18]: Let's also introduce method addSOAPHeader() and getSOAPHeader()

File: src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class SoapSampleTest implements IAbstractTest {
+
+    @Test
+    public void testAddInteger() {
+        AddIntegerMethod soap = new AddIntegerMethod();
+        soap.setProperties("api/soap/soap.properties");
+        soap.setHeaders(String.format("SOAPAction=%s", "http://tempuri.org/SOAP.Demo.AddInteger"));
```
Comment (szagriychuk-ua) [inline src/test/java/com/qaprosoft/carina/demo/SoapSampleTest.java:18]: that's better to move setHeaders() call into AddIntegerMethod as it's default action for this method

**Pull request merged**